### PR TITLE
version: revert my broken "optimization", redo it right

### PR DESCRIPTION
**Pull request by danderson**
Four commits: first two revert back to before I refactored everything into init(), second two reapply the MajorMinorPatch fix, and lazily memoize more string slicing work (mac/windows flavor reporting).

### CODE REVIEW

**Comment by danderson** [comment]
I plan to squash the commits into one on submit, but keeping them separate for easier review.

File: version-embed.go [hunk 1]
```diff
@@ -6,13 +6,8 @@ package tailscaleroot
...
 import _ "embed"
 
-// VersionDotTxt is the contents of VERSION.txt. Despite the tempting filename,
-// this does not necessarily contain the accurate version number of the build, which
-// depends on the branch type and how it was built. To get version information, use
-// the version package instead.
-//
 //go:embed VERSION.txt
-var VersionDotTxt string
+var Version string
```
Comment (bradfitz) [inline version-embed.go:10]: did you mean to revert this part?

Comment (danderson) [reply version-embed.go:10]: Oops, no. Fixed.

File: version/version.go [hunk 2]
```diff
@@ -5,96 +5,30 @@
...
 func init() {
-	initVersion()
-	initUnstable()
-	initMiscTraits()
-}
-
-func initVersion() {
 	defer func() {
-		// Must be run after Short has been initialized, easiest way to do that
-		// is a defer.
```
Comment (bradfitz) [inline version/version.go:97]: you probably didn't mean to revert this comment?

File: version/version.go [hunk 3]
```diff
@@ -103,8 +37,6 @@ func initVersion() {
 		return
 	}
 
-	// Otherwise, make approximate version info using Go 1.18's built-in git
-	// stamping.
```
Comment (bradfitz) [inline version/version.go:107]: mean to delete this?

**Comment by danderson** [comment]
Put the mistakenly lost comments back and rechecked the whole diff for other omissions. All looks correct to me now. Going to squash, let CI run over the totality, then merge.

**Pull request merged**